File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nums_with_index = []
        for i, num in enumerate(nums):
            nums_with_index.append((num, i))

        nums_with_index.sort()
        s,e= 0, len(nums_with_index)-1
        while s<e:
            if nums_with_index[s][0] + nums_with_index[e][0] == target:
                return [nums_with_index[s][1],nums_with_index[e][1]]
            if nums_with_index[s][0] + nums_with_index[e][0] > target:
                e-=1
            else:
                s+=1
            
        return []

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numsWithIndex =  new HashMap<>();
        for (int i = 0; i<nums.length;i++){
            numsWithIndex.put(i,nums[i]);
        }
        List<Map.Entry<Integer,Integer>> lists = new ArrayList<>(numsWithIndex.entrySet());
        lists.sort(Map.Entry.comparingByValue());
        int s = 0;
        int e = nums.length-1;
        while(s<e){
            if (lists.get(s).getValue() + lists.get(e).getValue() == target){
                return new int[] {lists.get(s).getKey(),lists.get(e).getKey()};
            }
            if(lists.get(s).getValue() + lists.get(e).getValue() > target){
                e-=1;
            }
            else{
                s+=1;
            }
        }
        return new int[] {};
    }
}